example/ExampleMain.cpp

```diff
@@ -72,12 +91,40 @@ std::unique_ptr<Module> createModuleExample(LLVMContext &context) {
 }
 
 int main(int argc, char **argv) {
+  llvm::cl::ParseCommandLineOptions(argc, argv);
+
   LLVMContext context;
   auto dialectContext = DialectContext::make<xd::ExampleDialect>(context);
 
-  auto module = createModuleExample(context);
-
-  module->print(llvm::outs(), nullptr, false);
+  if (g_action == Action::Build) {
+    auto module = createModuleExample(context);
+    module->print(llvm::outs(), nullptr, false);
+  } else if (g_action == Action::Verify) {
+    if (g_inputs.size() != 1) {
+      errs() << "Need exactly one input module\n";
+      return 1;
+    }
+
+    auto fileOrErr = MemoryBuffer::getFileOrSTDIN(g_inputs[0]);
+    if (std::error_code errorCode = fileOrErr.getError()) {
+      errs() << "Could not open input file " << g_inputs[0] << ": "
+             << errorCode.message() << '\n';
+      return 1;
+    }
+
+    SMDiagnostic error;
+    std::unique_ptr<Module> module = parseAssembly(**fileOrErr, error, context);
+    if (!module) {
+      error.print("llvm-dialects-example", errs());
+      errs() << "\n";
+      return 1;
+    }
+
+    if (!verify(*module, outs()))
+      return 1;
+  } else {
+    llvm_unreachable("unhandled action");
```

Comment:
Maybe use `report_fatal_error` instead? Per my understand, `llvm_unreachable` does not guarantee error messages for the benefit of better performance, but that should not matter here.

Reply:
This is more like an assertion because the enum of actions should be closed. But it doesn't really matter.